testonly/hammer/hammer.go

```diff
@@ -181,15 +185,58 @@ func HitMap(cfg MapConfig) error {
 		}
 	}(ticker.C)
 
-	for count := uint64(1); count < cfg.Operations; count++ {
-		if err := s.retryOneOp(ctx); err != nil {
-			return err
+	var wg sync.WaitGroup
+	errs := make(chan error, cfg.NumCheckers)
+
+	cctx, cancel := context.WithCancel(ctx)
+	for i := 0; i < cfg.NumCheckers; i++ {
+		wg.Add(1)
+		go func(i int) {
+			glog.Infof("%d: start checker %d", s.cfg.MapID, i)
+			defer wg.Done()
+			// Note that we deliberately don't use the cancelable context
+			// cctx so that cancellation in mid-request doesn't give internal
+			// warnings.
+			err := s.readChecker(ctx, cctx.Done())
+			if err != nil {
+				errs <- err
+			}
+			glog.Infof("%d: checker %d done with %v", s.cfg.MapID, i, err)
```

Comment:
Should this return a `nil` err when done in order to have a predictable number of total errors in the channel ?
`errs <- nil` ?

Reply:
I don't think it needs to be predictable, just bounded (to which end I've added +1 to allow for the main goroutine).

Reply:
Sounds good